### PR TITLE
Describe objective section in root README

### PR DESCRIPTION
**Pull request by meaboutsoftware**
This PR includes the description of objective section and its subchapters:

Problem - explaining why we decided for creating such a repository
Proposed Solution - explaining how we would like to tackle the problem down

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -1,10 +1,62 @@
...
+- Loose coupling
+- Architecture decision log
+- Good coding practices
+
+What we do not focus on:
+
+- Frontend (you can use React, Vue, Angular, Svelte or anything else)
+- Logging (you can use https://serilog.net/[Serilog])
+- Request validation (you can use https://docs.fluentvalidation.net/en/latest[Fluent Validation])
+- Contract testing (you can use https://github.com/VerifyTests/Verify[Verify])
```
Comment (kamilbaczek) [inline README.adoc:54]: I haven't seen verify used to to contract testing. I suggest recommend pact-net to this type of tests
```suggestion
- Contract testing (you can use https://github.com/pact-foundation/pact-net[PactNet])
```

File: README.adoc [hunk 2]
```diff
@@ -1,10 +1,62 @@
...
+
+**Chapter 1:** Initial architecture
+
+**Chapter 2:** Projects Separation
+
+**Chapter 3:** Microservice Extraction
+
+**Chapter 4:** Domain Model
+
+In each chapter we describe the evolution of the architecture of a selected domain. We apply everything we have learned over many years of software development - our goal is to help you avoid making the same mistakes we have. This way you can be sure that your architecture will not be too trivial for too long, or too complex from the start. Treat it as a guide, something you can come back to as often as you like.
```
Comment (kamilbaczek) [inline README.adoc:34]: Our repo name is "Evolutionary Architecture". I think will be nice to mention about Architecture evolution here. What you think?

**Review by kamilbaczek — CHANGES_REQUESTED** [review]
Really good job with this description!

File: README.adoc (hunk 1 above)
Comment (meaboutsoftware) [reply README.adoc:54]: Verify is not that popular for this type of testing but works pretty well :)

We can suggest the most popular option, no problem

Comment (kamilbaczek) [reply README.adoc]: Please send me in DM, examples if you have :D I'm curious how this works :D

File: README.adoc (hunk 2 above)
Comment (meaboutsoftware) [reply README.adoc]: Done

**Pull request merged**